### PR TITLE
[FIX] generate cert functionality

### PR DESCRIPTION
**Pull request by Bobbins228**
# Issue link

# What changes have been made

Added function `get_secret_name` Which uses a label selector to gather the secret name for the given Ray Cluster.
# Verification steps

### Notebook server ODH/RHOAI/Local
* Clone this repository with `git clone https://github.com/project-codeflare/codeflare-sdk.git`
* Checkout this PR's branch
* Run `poetry build` - install if needed (`pip install poetry`)
* Run `pip install --force-reinstall dist/codeflare_sdk-0.0.0.dev0-py3-none-any.whl` 
* Restart your notebook kernel
## Testing
* Create a deployment of CodeFlare Operator from main 
* Run the basic interactive and local interactive demos 
* ensure that secrets are created when using `generate_cert()`
## Checks
- [ ] I've made sure the tests are passing. 
- Testing Strategy
   - [ ] Unit tests
   - [ ] Manual tests
   - [ ] Testing is not required for this change

### CODE REVIEW

File: src/codeflare_sdk/utils/generate_cert.py [hunk 1]
```diff
@@ -74,6 +75,24 @@ def generate_ca_cert(days: int = 30):
...
 
+def get_secret_name(cluster_name, namespace, api_instance):
+    label_selector = f"ray.openshift.ai/cluster-name={cluster_name}"
+    try:
+        secrets = api_instance.list_namespaced_secret(
+            namespace, label_selector=label_selector
+        )
+        for secret in secrets.items:
+            if (
+                f"{cluster_name}-ca-secret-" in secret.metadata.name
```
Comment (sutaakar) [inline src/codeflare_sdk/utils/generate_cert.py:86]: Would it have sense to rather select secret based on `ray.openshift.ai/cluster-name` label?

Comment (astefanutti) [reply src/codeflare_sdk/utils/generate_cert.py:86]: It seems, but it's true there can still be multiple secrets for the same cluster?

Comment (sutaakar) [reply src/codeflare_sdk/utils/generate_cert.py:86]: Though, there may be more secrets with the label.

Comment (Bobbins228) [reply src/codeflare_sdk/utils/generate_cert.py:86]: The oauth config secret also shares this label so we need to be more specific

Comment (astefanutti) [reply src/codeflare_sdk/utils/generate_cert.py:86]: At a later point, we can add an annotation to the client Secret to tell them apart more robustly.

Comment (sutaakar) [reply src/codeflare_sdk/utils/generate_cert.py:86]: The label selector is already there, I overlooked it.

**Review by sutaakar — APPROVED** [review]

**Review by astefanutti — APPROVED** [review]

**Pull request merged**